auto_dev/behaviours/protocol_scaffolder.py

```diff
@@ -0,0 +1,913 @@
+"""Protocol scaffolder."""
```

Comment:
This whole file I simply moved here because the `BehaviourScaffolder` depends on the `ProtocolScaffolder` as a base class.